apps/src/lab2/views/components/AiTutor2UI.tsx

```diff
@@ -0,0 +1,114 @@
+import React, {useCallback, useEffect, useRef} from 'react';
+
+import ChatMessage from '@cdo/apps/aiComponentLibrary/chatMessage/ChatMessage';
+import {Role} from '@cdo/apps/aiComponentLibrary/chatMessage/types';
+import UserMessageEditor from '@cdo/apps/aiComponentLibrary/userMessageEditor/UserMessageEditor';
+import AiTutor2Manager from '@cdo/apps/lab2/ai/AiTutor2Manager';
+import {useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+import moduleStyles from './AiTutor2UI.module.scss';
+
+interface AiTutor2UIProps {
+  allowChat?: boolean;
+  type: 'user' | 'hint';
+  question?: string;
+  getFullQuestionFromQuestion?: (
```

Comment:
nit: `getFullQuestionFromQuestion` does feel a bit wordy 🙂 maybe `getFullPrompt` or `formatQuestion` / `formatPrompt` or similar?

Reply:
I had a feeling!  I like `getFullPrompt` (and `getAiTutor2FullPrompt` for Python Lab).